assets/app/scripts/controllers/overview.js

```diff
@@ -231,14 +233,43 @@ angular.module('openshiftConsole')
       Logger.log("deployments (subscribe)", $scope.deployments);
     }));
 
-    // Sets up subscription for images and imagesByDockerReference
-    watches.push(DataService.watch("images", $scope, function(images) {
-      $scope.images = images.by("metadata.name");
-      $scope.imagesByDockerReference = images.by("dockerImageReference");
-      Logger.log("images (subscribe)", $scope.images);
-      Logger.log("imagesByDockerReference (subscribe)", $scope.imagesByDockerReference);
+    // Sets up subscription for imageStreams
+    watches.push(DataService.watch("imageStreams", $scope, function(imageStreams) {
+      $scope.imageStreams = imageStreams.by("metadata.name");
+      buildDockerRefMapForImageStreams();
+      fetchReferencedImages();
+      // TODO look for "referenced images"
+      Logger.log("imageStreams (subscribe)", $scope.imageStreams);
     }));
 
+    var fetchReferencedImages = function() {
```

Comment:
since this is asynchronous, make it return a promise that gets fulfilled when the fetching is done, in case anything needs to do follow-up work